src/cursor-doc/model.ts

```diff
@@ -224,7 +224,7 @@ export class LineInputModel implements EditableModel {
     const seen = new Set<number>();
     this.dirtyLines.sort();
     while (this.dirtyLines.length) {
-      let nextIdx = this.dirtyLines.shift();
+      let nextIdx = this.dirtyLines.shift() as number;
```

Comment:
Again, TypeScript isn't smart enough to figure out that the length check on the previous line means this will never be undefined and will always be a number.